app/merginapi.cpp

```diff
@@ -594,6 +635,66 @@ bool MerginApi::extractProjectName( const QString &sourceString, QString &projec
   }
 }
 
+std::shared_ptr<MerginProject> MerginApi::getProject( const QString &projectFullName )
+{
+  for ( std::shared_ptr<MerginProject> project : mMerginProjects )
+  {
+    if ( projectFullName == getFullProjectName( project->projectNamespace, project->name ) )
+    {
+      return project;
+    }
+  }
+
+  return std::shared_ptr<MerginProject>();
+}
+
+QString MerginApi::forceCreateDir( QString path )
```

Comment:
I would suggest to rename this function to something easier to understand, e.g. findUniqueProjectDirectorName( dir, name ). I would remove mkdir() calls, so it only comes up with good new directory name. The caller function would first get unique dir name, then call mkdir() itself.

Also please add docs what the function does.